packages/core/src/request-builder/header-builder/csrf-token-header.ts

```diff
@@ -93,11 +91,20 @@ export async function addCsrfTokenAndCookies<
   request: ODataRequest<RequestT>,
   headers: MapType<string>
 ): Promise<MapType<string>> {
+  if (!request.destination) {
+    throw Error('The request destination is undefined.');
+  }
   if (
     request.config.method === 'get' ||
     Object.keys(headers).includes('x-csrf-token')
   ) {
     return headers;
   }
-  return { ...(await buildCsrfHeaders(request, headers)), ...headers };
+  return {
+    ...(await buildCsrfHeaders(request.destination, {
+      headers,
+      url: request.relativeServiceUrl()
```

Comment:
Does this the right thing? The service paths does contain the host and so on like the destination? Or is this merged somewhere? I see a test in the csrf-token-header.spec.ts in the deprecated block, but I do not see where this is mocked.

Reply:
Yes, this is correct. 
`buildCsrfHeaders` uses the destination as a basis and builds the url on its own.